File: app/src/main/java/com/zack/enderweather/domain/activity/BaseActivity.java
package com.zack.enderweather.domain.activity;

import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;

public abstract class BaseActivity extends AppCompatActivity {

    /** Set up the {@link android.app.ActionBar}, if the API is available. */
    protected void setupActionBar() {
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            // Show the Up button in the action bar.
            actionBar.setDisplayHomeAsUpEnabled(true);
        }
    }
}

File: app/src/main/java/net/zackzhang/code/haze/view/activity/BaseActivity.kt
package net.zackzhang.code.haze.view.activity

import android.support.annotation.StringRes
import android.support.v7.app.AppCompatActivity
import android.widget.Toast

abstract class BaseActivity : AppCompatActivity() {

    protected fun setupActionBar() {
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
    }

    fun showToast(@StringRes msgResId: Int) {
        Toast.makeText(this, msgResId, Toast.LENGTH_SHORT).show()
    }

    open fun exit() {
        finish()
    }
}
